llmclient/llms.py

```diff
@@ -596,12 +598,22 @@ async def acompletion(self, messages: list[Message], **kwargs) -> list[LLMResult
                 output_messages = [Message(**completion.message.model_dump())]
 
             reasoning_content = None
-            if hasattr(completion.message, "provider_specific_fields"):
+            if (
+                hasattr(completion.message, "provider_specific_fields")
+                and completion.message.provider_specific_fields is not None
+            ):
+                # DeepSeek's reasoning:
```

Comment:
```suggestion
                # NOTE: DeepSeek's reasoning was added in:
```

I was not sure why you linked a closed issue, let's just give more context

Reply:
I wanted to keep track of the discussion. Would it be better to just remove the link and just keep the note?

Reply:
I think it's fine to link the ticket where it got added, just was asking for more context in the comment on why we have it is all